pages/dao/[symbol]/proposal/components/instructions/Mango/MangoV4/EditToken.tsx

```diff
@@ -167,6 +222,64 @@ const EditToken = ({
     )
     // eslint-disable-next-line react-hooks/exhaustive-deps -- TODO please fix, it can cause difficult bugs. You might wanna check out https://bobbyhadz.com/blog/react-hooks-exhaustive-deps for info. -@asktree
   }, [form])
+  useEffect(() => {
+    const getTokens = async () => {
+      const client = await getClient(connection, wallet!)
+      const group = await client.getGroup(GROUP)
+      const currentTokens = [...group.banksMapByMint.values()].map((x) => ({
+        name: x[0].name,
+        value: x[0].mint,
+      }))
+      setMangoGroup(group)
+      setTokens(currentTokens)
+    }
+    if (wallet?.publicKey) {
+      getTokens()
+    }
+  }, [connection && wallet?.publicKey?.toBase58()])
+  useEffect(() => {
+    if (form.token && mangoGroup) {
+      const currentToken = mangoGroup!.banksMapByMint.get(
+        form.token.value.toBase58()
+      )![0]
+      const vals = {
+        ...form,
+        oraclePk: currentToken.oracle.toBase58(),
+        oracleConfFilter: currentToken.oracleConfig.confFilter.toNumber(),
+        mintPk: currentToken.mint.toBase58(),
+        name: currentToken.name,
+        adjustmentFactor: currentToken.adjustmentFactor.toNumber(),
+        util0: currentToken.util0.toNumber(),
+        rate0: currentToken.rate0.toNumber(),
+        util1: currentToken.util1.toNumber(),
+        rate1: currentToken.rate1.toNumber(),
+        maxRate: currentToken.maxRate.toNumber(),
+        loanFeeRate: currentToken.loanFeeRate.toNumber(),
+        loanOriginationFeeRate: currentToken.loanOriginationFeeRate.toNumber(),
+        maintAssetWeight: currentToken.maintAssetWeight.toNumber(),
+        initAssetWeight: currentToken.initAssetWeight.toNumber(),
+        maintLiabWeight: currentToken.maintLiabWeight.toNumber(),
+        initLiabWeight: currentToken.initLiabWeight.toNumber(),
+        liquidationFee: currentToken.liquidationFee.toNumber(),
+        stablePriceDelayIntervalSeconds:
+          currentToken.stablePriceModel.delayIntervalSeconds,
+        stablePriceDelayGrowthLimit:
+          currentToken.stablePriceModel.delayGrowthLimit,
+        stablePriceGrowthLimit: currentToken.stablePriceModel.stableGrowthLimit,
+        minVaultToDepositsRatio: currentToken.minVaultToDepositsRatio,
+        netBorrowLimitPerWindowQuote: currentToken.netBorrowLimitPerWindowQuote.toNumber(),
+        netBorrowLimitWindowSizeTs: currentToken.netBorrowLimitWindowSizeTs.toNumber(),
+        borrowWeightScaleStartQuote: currentToken.borrowWeightScaleStartQuote,
+        depositWeightScaleStartQuote: currentToken.depositWeightScaleStartQuote,
+        groupInsuranceFund: false,
```

Comment:
what is best way to get it for edited token @microwavedcola1

Reply:
every token has a corresponding mint info, their token index fields should match, the mint info has a group insurance fund